src/AppBundle/Document/User.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * This is a Document class as per http://symfony.com/doc/current/bundles/DoctrineMongoDBBundle/index.html,
+ * built on the class provided by http://symfony.com/doc/master/bundles/FOSUserBundle/index.html
+ */
+
+namespace AppBundle\Document;
+
+use FOS\UserBundle\Model\User as BaseUser;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as MongoDB;
+
+/**
+ * @MongoDB\Document
+ */
+class User extends BaseUser
+{
+    //Uses native generated object id and represents as string.
+    /**
+     * @MongoDB\Id(strategy="auto")
+     */
+    protected $id;
+
+    /**
+     * User constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Return the id as string.
+     * @return string
+     */
+    public function getIdString()
+    {
+        return '' . $this->id;
+    }
```

Comment:
I indexed it as a string just for debug purposes, I was looking at it in the mongodb console, and I couldn't get the find query working successfully with a object id as the filter. I would've preferred to keep it as the native object id class.